python/foglamp/services/south/server.py

```diff
@@ -252,17 +252,27 @@ def run(self):
     async def shutdown(self, request):
         """implementation of abstract method form foglamp.common.microservice.
         """
-        _LOGGER.info('Stopping South Service plugin {}'.format(self._name))
+        async def do_shutdown(loop):
+            _LOGGER.info('Stopping South Service plugin {}'.format(self._name))
+            try:
+                await self._stop(loop)
+                self.unregister_service_with_core(self._microservice_id)
+            except Exception as ex:
+                _LOGGER.exception('Error in stopping South Service plugin {}, {}'.format(self._name, str(ex)))
+                raise web.HTTPInternalServerError(reason=str(ex))
+
+        def schedule_shutdown(loop):
+            asyncio.ensure_future(do_shutdown(loop), loop=loop)
+
         try:
-            await self._stop(asyncio.get_event_loop())
-            self.unregister_service_with_core(self._microservice_id)
+            loop = asyncio.get_event_loop()
+            loop.call_later(1, schedule_shutdown, loop)
```

Comment:
should not we use call_soon here?

Reply:
Why 'call_soon' here? loop is already running and we want to delay execution by some time.

Reply:
call_soon will also schedule a future a task, dealy is NOT required here. we can let event loop to handle it.

Reply:
call_soon schedules the task at the next immediately available time and this is the default mode but as discussed, we want some delay before task is fired. Hence call_later and that too after 1 sec only. Otherwise, we may experience 'shutdown completed' message appearing before 'shutdown is scheduled' message.